test/commandLine/commandLineTest.js

```diff
@@ -0,0 +1,662 @@
+
+    testApp.on('message', params => {
+      assert.equal(params.htmlValidator.enable, false)
+      done()
+    })
+  })
+
+  it('should change the app to disable validator ("--raw")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['--raw'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.htmlValidator.enable, false)
+      done()
+    })
+  })
+
+  it('should change the app to disable validator ("-r")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-r'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.htmlValidator.enable, false)
+      done()
+    })
+  })
+
+  it('should change the app to detach the html Validator ("detach-validator")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['detach-validator'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.htmlValidator.separateProcess, true)
+      done()
+    })
+  })
+
+  it('should change the app to detach the html Validator ("--background-validator")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['--background-validator'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.htmlValidator.separateProcess, true)
+      done()
+    })
+  })
+
+  it('should change the app to detach the html Validator ("-b")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-b'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.htmlValidator.separateProcess, true)
+      done()
+    })
+  })
+
+  it('should change the app to attach the html Validator ("attach-validator")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-attach-validator'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.htmlValidator.separateProcess, false)
+      done()
+    })
+  })
+
+  it('should change the app to attach the html Validator ("--attach-validator")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['--attach-validator'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.htmlValidator.separateProcess, false)
+      done()
+    })
+  })
+
+  it('should change the app to attach the html Validator ("-a")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-a'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.htmlValidator.separateProcess, false)
+      done()
+    })
+  })
+
+  /* check combination of flags */
```

Comment:
Since the first group of tests prove the functionality of each flag along with each alias we don't need to be quite so verbose in combination tests. Testing combinations with a single version of each flag will suffice.

Example: If we test `--development-mode --enable-validator --attach-validator` we don't need to also test `--dev --html-validator -a`.

Instead let's cut it down to just unique combinations.

This applies to mutually exclusive flags as well.